modules/mauth-authenticator/src/main/java/com/mdsol/mauth/RequestAuthenticator.java

```diff
@@ -108,7 +106,11 @@ public boolean authenticate(MAuthRequest mAuthRequest) {
 
     PublicKey clientPublicKey = clientPublicKeyProvider.getPublicKey(mAuthRequest.getAppUUID());
     if (mAuthRequest.getMauthVersion().equals(MAuthVersion.MWSV2)) {
-      return validateSignatureV2(mAuthRequest, clientPublicKey);
+      boolean isValidated = validateSignatureV2(mAuthRequest, clientPublicKey);
+      if (!isValidated && !v2OnlyAuthenticate) {
+        isValidated = fallbackValidateSignatureV1(mAuthRequest, clientPublicKey);
+      }
+      return isValidated;
```

Comment:
Please avoid boolean inversion as much as possible. It's easy to introduce bugs that way.

Do you think this is simpler?
```
if (v2OnlyAuthenticate) {
  return v2IsValidated
}
else {
  return v2IsValidated | fallbackValidateSignatureV1(mAuthRequest, clientPublicKey);
}
```

Reply:
fallbackValidateSignatureV1(...) should be called when v2IsValidated is false. To avoid boolean inversion, we can change as below 
```
if (v2OnlyAuthenticate) {
  return v2IsValidated;
}
else if (!v2IsValidated) {
  return fallbackValidateSignatureV1(mAuthRequest, clientPublicKey);
}
return v2IsValidated;

```

Reply:
Good point. **fallbackValidateSignatureV1**(...) should be called when v2IsValidated is false and not  v2OnlyAuthenticate. how do you think if simplify as below?
```
if (v2OnlyAuthenticate || v2IsValidated) {
  return v2IsValidated
}
else {
  return fallbackValidateSignatureV1(mAuthRequest, clientPublicKey);
}
```

Reply:
I want the code to follow the actual business logic as close as possible.

This is what I think the logic (in english) is:

```
If (v2OnlyAuthenticate)
  return result of v2 authentication
else
  return the result of v2 authentication, if that fails then return the result of v1 authentication
```

The code follows the logic

Reply:
updated as suggested https://github.com/mdsol/mauth-jvm-clients/commit/cbc0eca0e562d0b7b87d895e09e114f752da3b2a

Reply:
Thanks.